notes/2018-02-01.md

```diff
@@ -0,0 +1,220 @@
+
+  Ideally should be available for introspection. 
+
+  **Summary**:
+
+  - That PR needs some more time/energy from our discussion
+  - Needs to support old schemas
+  - Most scalars are custom
+
+5. ## Discuss subscriptions for live queries
+
+  Artsy use case: Live bidding platform for auctions. As a participant you see a real-time sub-second latency, updates can happen to many lots of hundreds. Current WS. The socket makes assumptions about what the client is interested in.
+
+  Subscriptions could be what is needed for live queries. The root field of a subscription would be the live query. Then subsequent queries can get delta updates. Connecting could include a timestamp to show when to send live data from, to handle reconnects.
+
+  Needs a way to decouple the update from the shape of the query. A scalar/obj could handle with nullability, but how do you deal with arrays, or deep hierarchies?
+
+  C: FB has done similar things, and bailed. It’s too tough to go from 1 query and handle the live changes.
+
+  C: We do polling live queries, keep sending the full payload. Fixes the array patching, or missed deltas. Not optimal. Saw serious issues with server resources with connections to dbs etc.
+
+  C: Event based subscriptions are what we’re talking about publicly. They can have tiny sub queries. So a new bid comes in we can make the query for highest bidder. This has been more robust. 
+
+  C: No right answer. 
+
+  C: Tried the same thing, hit the same roadblocks. Ended up with event subscriptions. Super hard. 
+
+  C: A lot of this is challenges of implementation, complexity may be specific to a backend so spec changes might not make sense.
+
+  C: For clients change patches should ideally be invisible, can move it to client-side transport.
+
+  C: Need for a layer between there.
+
+  C: Could work like keyframes do in MPEG, which send the full data every so often. A video is similar in its problems. 
+
+  C: What does it take to find some space to play with an idea like ^
+
+  C: Need more examples of it actually working, so we can talk about tradeoffs. 
+
+  Action items: 
+
+  - -Keep thinking about this
+
+  6. ## Discuss moving connection/global ID specs from Relay into best practices repo
+
+  Found it hard to encourage all developers to use connections, ID because of it’s link to relay.
+
+  Desire to move the Relay spec into a best practices section. Want to decouple connections from relay specifically. Global ID and connections are a great pattern.
+
+  C: Want to be cautious about this slowing to adoption for graphql?
+  Questions:
+
+  - -ID field make sense?
+  - -Does the Relay ID make sense?
+  - -Does pagination make sense?
+
+  - In favour of splitting into more multiple sections:
+
+   Cursor per item feels weird to be normative
+   Leeb has a doc that is basically this, which has other alternatives to connections
+   C: Perhaps “pattern” as a term, is a best option?
+   Relay modern doesn’t use the global ID
```

Comment:
This isn’t accurate. More accurate is that it uses an `id` field if it exists, otherwise generates its own IDs

For single nodes https://github.com/facebook/relay/blob/104b2c3a1e5199d7b2ba5c6e4ce0adffa6461fad/packages/relay-runtime/store/RelayResponseNormalizer.js#L259 and for plural nodes https://github.com/facebook/relay/blob/104b2c3a1e5199d7b2ba5c6e4ce0adffa6461fad/packages/relay-runtime/store/RelayResponseNormalizer.js#L308 (edited)